architecture/adr-016-shell-api.md

```diff
@@ -0,0 +1,77 @@
+<!-- Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. -->
+
+| Type          | ID            | Status        | Title                   |
+| ------------- | ------------- | ------------- | ----------------------- |
+| ADR           | adr-016       | Approved      | Shell API               |
+
+# Architecture Decision Record: Shell API
+
+## Context
+
+We wish to ensure apps written for ComposeUI can run in any shell implementation and provide a consistent developer experience across shells. In order to achieve this, a fixed set of features for shell interaction need to be defined and made available for developers.
+
+Another consideration we must make is how easy is it to bring existing applications into the ComposeUI environment. "Basic" web apps are expected to run out of the box, but apps targeted at running inside containers might leverage specific APIs. We need to consider API compatibility with other desktop containers as well as DesktopJS, an abstraction layer over multiple web container APIs we are maintaining.
+
+### A list of features we want to provide with high priority:
+#### Window management
+Opening and closing of child windows with the opportunity to customize their properties like size, position, available controls, "topmostness" etc.
+Child windows will need to fit the style of the Shell.
+Initially windows opened by modules are considered to be only part of the app state and thus will not be saved as part of the layout. In the future interaction with layout saving will need to be considered.
```

Comment:
> Initially windows opened by modules are considered to be only part of the app state and thus will not be saved as part of the layout.

In [ADR-013](https://github.com/morganstanley/ComposeUI/blob/main/architecture/adr-013-script-preloading-security.md) (which is in Draft) the decision was that
> ... popup windows and iframes will not inherit the parent window's module identity or injected scripts. When a module wants to pop up another window, it has to do so using container APIs (eg. FDC3 raiseIntent or open, or the Module Loader start API).

This to me means that windows opened by modules are new apps, cannot be part of the app's state which opened them. Am I missing something?